numpy/core/function_base.py

```diff
@@ -87,8 +87,10 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     div = (num - 1) if endpoint else num
 
     # Convert float/complex array scalars to float, gh-3504
-    start = start * 1.
-    stop = stop * 1.
+    # Make sure one can use variables that have an __array_interface__, gh-6634
+    from numpy import asarray
```

Comment:
Please use `asanyarray` here, so that numpy subclasses get passed through (`np.linspace` works with `Quanity` right now; this would break it). Note also that you can just add this import to the ones imported from `numeric` at the top. 

Actually, the multiplication with `1.` is not necessary anymore if you just do
```
start = asanyarray(start, dtype=np.float64)
```